src/query/storage/cache/redis_cache.go

```diff
@@ -1,26 +1,37 @@
 package cache
 
 import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
 	"time"
 
+	"github.com/m3db/m3/src/query/block"
+	"github.com/m3db/m3/src/query/generated/proto/prompb"
+	"github.com/m3db/m3/src/query/storage"
 	radix "github.com/mediocregopher/radix/v3"
 
 	"go.uber.org/zap"
 )
 
-type RedisCache struct {
-	client       radix.Client
-	redisAddress string
-	logger       *zap.Logger
-}
-
 const (
+	// Time that keys expire after (in seconds)
+	ExpirationTime string = "300"
+	// Blank result used to denote an empty PromResult
+	EmptyResult string = "{}"
 	// Minimum number of connections pools to Redis to keep open
 	MinPools int = 10
 	// Time to wait before creating a new Redis connection pool (in ms)
 	CreateAfterTime time.Duration = 100 * time.Millisecond
 )
 
+type RedisCache struct {
```

Comment:
Can we put a dynamic config to enable it or disable this cache? This will be easier for your to write an integration tests.

Reply:
Added a new HTTP endpoint to disable cache